plugins/google/pkg/googleclient/google.go

```diff
@@ -0,0 +1,171 @@
+package googleclient
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	admin "google.golang.org/api/admin/directory/v1"
+	"google.golang.org/api/option"
+)
+
+type GoogleClient struct {
+	googleClient *admin.Service
+}
+
+func GetRefreshToken(ctx context.Context, clientID, clientSecret string, port int) (string, error) {
+	redirectURL := fmt.Sprintf("http://localhost:%d", port)
+	config := &oauth2.Config{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
+		RedirectURL:  redirectURL,
+		Scopes: []string{
+			admin.AdminDirectoryUserScope,
+			admin.AdminDirectoryGroupScope,
+		},
+		Endpoint: google.Endpoint,
+	}
+
+	// Create an HTTP server for handling the OAuth 2.0 callback
+	server := &http.Server{Addr: fmt.Sprintf(":%d", port), ReadHeaderTimeout: 5 * time.Second}
+
+	// Create a channel to receive the authorization code
+	authCodeChan := make(chan string)
+
+	// Start the HTTP server in a separate goroutine
+	go func() {
+		http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
There's no reason to create the handler inside a go routine. The only thing that needs to run in the background is the call to `server.ListenAndServer()`.